prime-router/src/test/kotlin/metadata/LivdMapperTests.kt

```diff
@@ -253,4 +256,317 @@ class LivdMapperTests {
         assertThat(mapper.apply(testKitElement, emptyList(), listOf(ev2, modeT)).value).isEqualTo(testKitIdBinax)
     }
+
+    /**
+     * The fake LIVD data to be used in the table
+     */
+    private val fakeLivdTableData = listOf(
+        listOf(
+            LivdTableColumns.MODEL.colName, LivdTableColumns.TEST_PERFORMED_CODE.colName,
+            LivdTableColumns.TESTKIT_NAME_ID.colName, LivdTableColumns.EQUIPMENT_UID.colName,
+            LivdTableColumns.PROCESSING_MODE_CODE.colName
+        ),
+        // Rows 1-3: Similar to 1copy COVID-19 qPCR Multi Kit, the only change is the equipment UID
+        listOf("model1", "90001-1", "model1kit1", "model1uid1", ""),
+        listOf("model1", "90001-1", "model1kit1", "model1uid2", ""),
+        listOf("model1", "90001-1", "model1kit1", "model1uid3", ""),
+
+        // Rows 4-6: Similar to TRUPCR SARS-CoV-2 Kit, with different LOINC codes
+        listOf("model2", "90001-1", "model2kit1", "model2uid1", ""),
+        listOf("model2", "90001-1", "model2kit1", "model2uid2", ""),
+        listOf("model2", "90001-2", "model2kit1", "model2uid3", ""),
+
+        // Rows: 7-8: Test devices 
+        listOf("model3", "90003-1", "model3kit1", "model3uid1", LIVDLookupMapper.testProcessingModeCode),
+        listOf("model4", "90003-1", "model4kit1", "model4uid1", LIVDLookupMapper.testProcessingModeCode),
+
+        // Rows 9-11: Some devices with similar data, but different test performed code
+        listOf("model5", "90005-1", "model5kit1", "model5uid1", ""),
+        listOf("model5", "90005-2", "model5kit1", "model5uid1", ""),
+        listOf("model5", "90005-3", "model5kit1", "model5uid1", ""),
+    )
+
+    private val livdTable = LookupTable("LIVD", fakeLivdTableData)
+
+    private val modelNameElement = Element(
+        ElementNames.EQUIPMENT_MODEL_NAME.elementName,
+        tableRef = livdTable,
+        tableColumn = LivdTableColumns.MODEL.colName
+    )
+    private val testKitElement = Element(
+        ElementNames.TEST_KIT_NAME_ID.elementName,
+        tableRef = livdTable,
+        tableColumn = LivdTableColumns.TESTKIT_NAME_ID.colName
+    )
+    private val equipmentUidElement = Element(
+        ElementNames.EQUIPMENT_MODEL_ID.elementName,
+        tableRef = livdTable,
+        tableColumn = LivdTableColumns.EQUIPMENT_UID.colName
+    )
+    private val deviceIdElement = Element(
+        ElementNames.DEVICE_ID.elementName,
+        tableRef = livdTable
+    )
+    private val testPerformedCodeElement = Element(
+        ElementNames.TEST_PERFORMED_CODE.elementName,
+        tableRef = livdTable,
+        tableColumn = LivdTableColumns.TEST_PERFORMED_CODE.colName
+    )
+    private val processingModeElement = Element(ElementNames.PROCESSING_MODE_CODE.elementName)
+
+    /**
+     * Grab a cell value from the lIVD data based on the [deviceindex] starting at 1 and a given [element].
```

Comment:
at line 318 Is "lIVD" supposes to be "LIVD" instead?

Reply:
Oops.  Fixed.